lib/plugins/aws/package/compile/events/sns/index.js

```diff
@@ -64,6 +64,9 @@ class AwsCompileSNSEvents {
                   if (topicArn.indexOf('arn:') === 0) {
                     const splitArn = topicArn.split(':');
                     topicName = splitArn[splitArn.length - 1];
+                    if (splitArn[3] !== this.options.region) {
+                      region = splitArn[3];
```

Comment:
🐛 Bug...

this line cuase issue when arn contain pseudo params:
e.g 
```
arn:aws:sns:${AWS::Region}:...
```

it will calculate region as `${AWS`

@Simon Elder  @herebebogans @pmeuns.

Reply:
https://github.com/svdgraaf/serverless-pseudo-parameters/issues/38

Yep, it breaks alright 😅 Not sure what the best cause of action is here. I think it's dirty to do a direct split like this. A pseudo parameter is a valid value for the Arn. Even just a `Ref` would be a valid, and that would break this too.

Perhaps check the found region with a set list of available regions from the aws-sdk? Or add a `region` parameter to set a specific region on the sns trigger?